hack/generator/pkg/codegen/pipeline_create_arm_types.go

```diff
@@ -198,22 +199,29 @@ func transformTypeDefinition(
 	def astmodel.TypeDefinition,
 	handlers []conversionHandler) (astmodel.TypeDefinition, error) {
 
-	originalType, ok := def.Type().(*astmodel.ObjectType)
-	if !ok {
-		return astmodel.TypeDefinition{}, errors.Errorf("input type %q (%T) was not of expected type Object", def.Name(), def.Type())
+	if !astmodel.IsObjectType(def.Type()) {
```

Comment:
This allows ARM types, which seems wrong since we don't want to allow ARM -> ARM transformations?

Reply:
Might be better to have an `IsObjectType` (used here) and an `IsObjectOrArmType` (used elsewhere where you want to treat them the same - not exactly sure where that is).

Reply:
Actually on further inspection I only see this used here - why have the duality at all then (Where an ARMType `IsObjectType`?)